### ISSUE TITLE
[Bug]: TextMeshPro and URP Additional Camera Data scripts aren't importing correctly.

### ISSUE THREAD

**Issue by StupidRepo** [AssetRipper/AssetRipper#1574]
### Are you on the latest version of AssetRipper?

Yes, I'm on the latest release of AssetRipper.

### Which release are you using?

Windows x64

### Which game did this occur on?

[Baldi's Basics Classic Remastered](https://steamdb.info/app/1712830/)

### Which Unity version did this occur on?

2020.3.38f1

### Is the game Mono or IL2Cpp?

Mono

### Describe the issue.

It seems all text objects appear as having a missing script. The Unity.TextMeshPro.dll is exported into Plugins, and I've even tried deleting that DLL and installing from package manager but that didn't help either :(

**Comment by StupidRepo** [comment]
Hm, seems like BB is mentioned a lot alongside TMP errors. It does indeed seem to be some kind of GUID issue.

**Comment by ds5678** [comment]
You probably have script errors.

**Comment by StupidRepo** [comment]
Nope, it is a GUID problem. I have found a project from you called GuidUpdater, might try it rn.

**Comment by ds5678** [comment]
> I've even tried deleting that DLL and installing from package manager but that didn't help either

Unity doesn't allow you to switch between using C# scripts and compiled DLL files.

**Comment by wyw666666** [comment]
I had the same problem, have you solved it yet?

**Comment by StupidRepo** [comment]
Sadly, no :(

**Comment by Scorpy-37** [comment]
Same problem here, all the TMP components break and become "Missing Script". I've tried installing TMP into an empty project and putting the Assets into the project and installing TMP while in safe-mode but no luck.

**Comment by StupidRepo** [comment]
I tried to copy the .meta files over from the decompiled game to a Unity project with the Package Manager TextMeshPro, and did get some components to show up however it then gave me a bunch of null errors in the output when I tried to change font, text, etc.

**Comment by Scorpy-37** [comment]
Did you do it before or after running the decompiled game inside the Unity Editor?

**Comment by 1-TB** [comment]
Im having the same issue.

**Comment by protogenreal** [comment]
The Issue Is Because Unity Doesn't Let You Use DLL's As C# The Best Method To Get It Back Is By Getting AssetRipper Premium 1.9 (or newer before i2lcpp extraction as it fills the dll's and I use it and its efficient) And Getting The AssetRipper Mining PackageData And Finding The TextMeshPro Json And Using That So AssetRipper Replaces The GUID's With The Proper One's.

**Comment by StupidRepo** [comment]
Unity does let you use DLLs as C#... kinda. If you have a compiled DLL, you can drop it into Unity and it'll load it's classes.

**Comment by bsoda77854** [comment]
use this lmao https://github.com/ningengit/GUIDcorrector

**Comment by AndresDevvv** [comment]
having same issue
